test/e2e/extensions/high_availability_test.go

```diff
@@ -41,18 +42,35 @@ func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
 
 	list, err := getAgoneseExtensionsPods(ctx)
 	logger.Infof("Length of pod list is %v", len(list.Items))
-	assert.NoError(t, err, "Could not get list of Extension pods")
+	for i := range list.Items {
+		logger.Infof("Name of extensions pod %v: %v", i, list.Items[i].ObjectMeta.Name)
+		logger.Infof("Host IP %v", list.Items[i].Status.HostIP)
+		logger.Infof("Pod IPs %v", list.Items[i].Status.PodIPs)
+	}
+	require.NoError(t, err, "Could not get list of Extension pods")
 	assert.Greater(t, len(list.Items), 1, "Cluster has no Extensions pod or has only 1 extensions pod")
 
-	logger.Info("Removing one of the Extensions Pods")
+	logger.Infof("Removing one of the Extensions Pods: %v", list.Items[1].ObjectMeta.Name)
 	deleteAgonesExtensionsPods(ctx, t)
+	require.Eventually(t, func() bool {
+		extList, err := getAgoneseExtensionsPods(ctx)
+		if err != nil {
+			return true
+		}
 
-	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+		if len(extList.Items) == 2 {
```

Comment:
A check I would add - is the len(2) but also is the `list` above different from the list we just got in `extList` - that way we definitely know we have a new replacement pod, and it's up and running.

Otherwise we could get 1 pod running 1 terminating.

Reply:
You could likely just do a comparison of sorted pod names in both lists. A bit of code, but then we know we're in the place we expect to be.

Reply:
I think this would be easier to integrate into `deleteAgonesExtensionsPods`, and here's why: that function is misnamed. It's deleting exactly one pod, and that pod has a name, known to that function as `list.Items[1].ObjectMeta.Name`.

It's much easier just to manually use the KubeClient.Delete, then require.Eventually() that the KubeClient.Get on the Pod fails - i.e. manually delete the single pod and just wait on that single pod to be gone.

And rename the function as `deleteAgonesExtensionPod`.

I would maybe then change `waitForAgonesExtensionsRunning` to actually make sure there are 2 running.

Reply:
Works for me 👍🏻

Reply:
This looks fine to me - it's more code than what I suggested above but it'll get the job done for sure. We can clean it up later if needed.